runtime/compiler/runtime/J9ValueProfiler.hpp

```diff
@@ -817,8 +817,8 @@ TR_HashTableProfilerInfo<T>::initialHashConfig(HashFunction &hash, T value)
 
    // A bit index hash should try to nominate one bit that is zero for all
    // A bit shift hash should try to nominate one bit that is zero for all and to the left of all
-   if (getHashType() == BitIndexHash && ~value != 0 ||
-       getHashType() == BitShiftHash && (~value >> 8) != 0)
+   if ((getHashType() == BitIndexHash && ~value != 0) ||
+       (getHashType() == BitShiftHash && (~value >> 8) != 0))
```

Comment:
Can you bracket each of the clauses individually as well?  It's clearer and doesn't require anyone to think about order of operations when reading the code

Reply:
Ping @dnakamura.  I think @DanHeidinga means something like:
```
   if ((getHashType() == (BitIndexHash && ~value !=0)) ||
       ((getHashType() == (BitShiftHash && (~value >> 8))) != 0))
```